src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -335,7 +334,7 @@ private function parseXmlValue(\DOMNode $node, array $context = array())
         }
 
         foreach ($value as $key => $val) {
-            if (is_array($val) && 1 === count($val)) {
+            if (is_array($val) && 1 === count($val) && (!(isset($context['as_collection']) && $context['as_collection']) || !is_array($val[0]))) {
```

Comment:
using `empty()` would make the code simpler I suppose
I don't understand the last condition `!is_array($val[0])` - it feels unrelated. Is it?

Reply:
I'm looking for it, `!is_array($val[0])` makes it possible to return the value directly instead of having two arrays: 

```
$expected = array(
            '@nodeType' => 'order_row',
            '@virtualEntity' => 'true',
            'order_row' => array(array(
                'id' => 16,
                'test' => 16,
            )),
        );
```
vs
```
$expected = array(
            '@nodeType' => 'order_row',
            '@virtualEntity' => 'true',
            'order_row' => array(array(
                'id' => array(16),
                'test' => array(16),
            )),
        );
```

Do you mean replacing 
```
is_array($val) && 1 === count($val)
```

to 

```
!empty($val)
```

it's not possible